### DISCUSSION TITLE
Questions about GitHub authentication

### DISCUSSION THREAD

**Discussion by rdbeach** [coder/code-server#2535] [General] [answered]
Will the code for the auth relay server be made public? And do you plan on maintaining this endpoint indefinitely?

```const AUTH_RELAY_SERVER = 'auth.code-server.dev';```

Also, I noticed that in the package 3.8.0 release, the lib/vscode/resources folder does not contain the web folder which is necessary for authentication. Is there a reason for this?

**Comment by code-asher** [comment, marked as answer]
I've got no problem making it public. Thoughts @nhooyr?

We should be packaging `callback.html` at least in
`lib/vscode/resources/web`. We copied the relevant code from
`code-web.js` into our own web server code since we can't just import it
as it would create a new web server.

**Reply by code-asher** [reply]
Also, current plans are to maintain it indefinitely unless VS Code's relay gains support for arbitrary domains by some miracle. 😄

**Reply by code-asher** [reply]
Maybe `callback.html` isn't getting bundled for some reason. :(
https://github.com/cdr/code-server/issues/1883#issuecomment-753741580

**Reply by peterholko-pingidentity** [reply]
@code-asher Are you still planning to maintain auth.code-server.dev ? 

It appears to be offline:

"Error: Page not found
The requested URL was not found on this server."

**Reply by code-asher** [reply]
Upstream in Code 1.66 the relay code was removed in favor of a flow where you copy-paste an authentication token.  That version was bundled into 4.4.0 of code-server, a little less than a year ago.  

Although while I think the copy-paste system is unfortunately inferior to our relay it is the status quo now and it also means Code has supported arbitrary domains for a while now so we thought it was about time to shut down the relay.

Is there something keeping you on an old version of code-server?
